editor/src/components/canvas/canvas-strategies/canvas-strategy-types.ts

```diff
@@ -49,13 +50,29 @@ export function strategyApplicationResult(
   }
 }
 
-export interface ControlWithKey {
-  control: React.FC<React.PropsWithChildren<unknown>>
+export interface ControlForStrategy<P> {
+  type: 'ControlForStrategy'
+  control: React.FC<P>
+}
+
+export function controlForStrategyMemoized<P>(control: React.FC<P>): ControlForStrategy<P> {
+  return { type: 'ControlForStrategy', control: React.memo<any>(control) }
```

Comment:
I'm pretty sure we can use `React.memo<P>` here

Reply:
<img width="899" alt="image" src="https://user-images.githubusercontent.com/2226774/194076193-066f31f0-80fe-49a3-b585-313ea504323d.png">

I was fucking around with it for like an hour before I gave up. This <any> doesn't leak outside of the function, so this is not affecting anything in practice. It looks silly but it isn't worth more effort trying to type this up

Reply:
Oh ffs